### PR TITLE
k8s spark operator compatibility

### LINKED ISSUE #32363: K8S Spark Operator doesn't delete an older application with the same name anymore
### Apache Airflow version

2.6.2

### What happened

Changes to the SparkKubernetesOperator which introduced in https://github.com/apache/airflow/pull/21092 were reverted by the changes added in https://github.com/apache/airflow/pull/29977

Until 2.6.2 when a spark application was created with the same name of a previous application, the old application (pods etc') was automatically deleted and the new application started (with the warning of 'Deleted SparkApplication with the same name' in the logs).
After 2.6.2 this scenario is causing an error starting the new application (409 - Already exists) and the previous application needs to be manually deleted (there is no easy way to delete it programmatically from airflow dags).

Was this behavior change done on purpose?
I can suggest to add a parameter to control the desired behavior.

### What you think should happen instead

Changes to the SparkKubernetesOperator which introduced in https://github.com/apache/airflow/pull/21092 were reverted by the changes added in https://github.com/apache/airflow/pull/29977

### How to reproduce

Start a new spark application with the same name of the previous one, compare 2.6.2 vs 2.5.3

### Operating System

EKS

### Versions of Apache Airflow Providers

_No response_

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by blcksrx**
Changes in this PR:
- Added `override_existing` argument with default to the operator.
- Uses the K8S application `uid` instead of naming

closes: #32363

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/cncf/kubernetes/operators/spark_kubernetes.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ class SparkKubernetesOperator(BaseOperator):
     def __init__(
         self,
         *,
-        application_file: str | dict,
```
Comment (josh-fell) [inline airflow/providers/cncf/kubernetes/operators/spark_kubernetes.py]: Just curious, would users have passed `application_file` as a `dict` before or did it just not work previously? I'm just thinking of any backwards compat issues here albeit potentially small.

Comment (blcksrx) [reply airflow/providers/cncf/kubernetes/operators/spark_kubernetes.py]: Ah, that's strange, cause I didn't change it.‌ sorry

**Comment by assaf-xm** [comment]
@blcksrx is there any progress with the review of this PR?
(this is a small PR)

**Comment by blcksrx** [comment]
@assaf-xm no review yet :(

**Comment by ruslanguns** [comment]
This PR is very important for us, we are start working with Airflow and Spark and this additional means a lot. I really encourage the reviewers to review this. Thanks in advanced:

FYI @jedcunningham @hussein-awala @josh-fell

**Comment by potiuk** [comment]
Can you rebase it please @blcksrx ? There are some conflicts.

**Comment by blcksrx** [comment]
@potiuk I see a lot of changes, It gonna check it during the weekened

**Comment by romsharon98** [comment]
I think it will be nice to add this `override_existing` as templated field.
If it's not to much can u add it?

**Comment by blcksrx** [comment]
A major change happened in the operator actually, that makes it hard to do it!
The new change generates a new job name(at least what i understood from the code) and it needs collaboration of these changes commiters

**Comment by romsharon98** [comment]
> A major change happened in the operator actually, that makes it hard to do it! The new change generates a new job name(at least what i understood from the code) and it needs collaboration of these changes commiters

isn't it just to add this line:
```python
    template_fields: Sequence[str] = ("application_file", "namespace", "override_existing")
```
or I missing something?

**Comment by blcksrx** [comment]
This part is simple,the operator it self got completely changed, check it from the main branch

**Pull request closed without merging**